python/across_api/base/fov.py

```diff
@@ -0,0 +1,382 @@
+
+    # Normalize it
+    prob = np.divide(prob, np.sum(prob))
+
+    return prob
+
+
+class FOVBase(ACROSSAPIBase):
+    visible_pixels: np.ndarray
+
+    def probability_in_fov(
+        self,
+        skycoord: Optional[SkyCoord] = None,
+        error_radius: Optional[u.Quantity] = None,
+        healpix_loc: Optional[FITS_rec] = None,
+    ) -> float:
+        """
+        For a given sky position and error radius, calculate the probability of
+        the sky position being inside the field of view (FOV).
+
+        Parameters
+        ----------
+        skycoord
+                SkyCoord object representing the position of the celestial object.
+        error_radius
+                The error radius for the sky position. If not given, the `skycoord`
+                will be treated as a point source.
+        healpix_loc
+                HEALPix map of the localization.
+        time
+                The time of the observation.
+        ephem
+                Ephemeris object.
+
+        """
+        # For a point source
+        if skycoord is not None and (error_radius is None or error_radius == 0.0):
+            in_fov = self.in_fov(skycoord)
+            return 1.0 if in_fov else 0.0
+        # For a circular error region
+        if skycoord is not None and error_radius is not None:
+            return self.in_fov_circular_error(
+                skycoord=skycoord, error_radius=error_radius
+            )
+        # For a HEALPix map
+        elif healpix_loc is not None:
+            return self.in_fov_healpix_map(healpix_loc=healpix_loc)
+
+        # We should never get here
+        raise AssertionError("No valid arguments provided")
+
+    def in_fov(
+        self,
+        skycoord: SkyCoord,
+    ) -> Union[bool, np.ndarray]:
+        """
+        Is a coordinate or set of coordinates `skycoord` inside the FOV and not
+        Earth occulted.
+
+        Note that this method only checks if the given coordinate is Earth
+        occulted, so defines a simple 'all-sky' FOV with no other constraints.
+        For more complex FOVs, this method should be overridden with one that
+        also checks if coordinate is inside the bounds of an instrument's FOV
+        for a given spacecraft attitude.
+
+        Parameters
+        ----------
+        skycoord
+                SkyCoord object representing the celestial object.
+        time
+                Time object representing the time of the observation.
+        ephem
+                Ephemeris object
+
+        Returns
+        -------
+        bool
+                True or False
+        """
+
+        # Check if skycoord pixels in list of visible pixels
+        skycoord_pix = hp.ang2pix(
+            hp.order2nside(HEALPIX_MAP_EVAL_ORDER),
+            skycoord.ra.value,
+            skycoord.dec.value,
+            lonlat=True,
+            nest=True,
+        )
+        in_fov = np.ma.isin(skycoord_pix, self.visible_pixels)
```

Comment:
[np.searchsorted](https://numpy.org/doc/stable/reference/generated/numpy.searchsorted.html) might do this faster because we can take advantage of `self.visible_pixels` being pre-sorted.

Reply:
I tested this out, and wasn't getting the same results as the masked array implementation in the tests.I also found that it seemed to be around the same speed as the masked array.

Reply:
If you want to use `isin`, then please use the ndarray version and not the matrix version:

```suggestion
        in_fov = np.isin(skycoord_pix, self.visible_pixels)
```

Here is a version with searchsorted.

```
def is_in_sorted_array(a, v):
    i = np.clip(np.searchsorted(a, v), 0, len(a) - 1)
    return a[i] == v
```

I would think that `np.isin` is O(N log N + M log N) where N and M are the lengths of the two arrays, because it must first sort the array. However if it uses a sufficiently clever sort then it might be only O(N + M log N) if the array is pre-sorted. However, the `is_in_sorted_array` is only O(M log N).

This is definitely a micro-optimization given the small sizes of these arrays.

Reply:
I tested both, and actually found that `np.isin()` is actually faster for our tests. Not by much, but it's less lines of code and more readable, so I went with that